GameDB: add missing serials, upscaling fixes, name corrections

noigeaR opened the pull request:
### Description of Changes
Add missing serials for:
- Hokka Hoka Sentou (SLPS-20217)
- Pachi-Slot Aruze Oukoku 7 (Disc 1) (Ekishou Disc)  (SLPS-20220)
- Million God (SLPS-20255)
- Yamasa Digi World 4DX (SLPS-20257)

Correct names based on redump.org database and www.jp.playstation.com/software/title/ pages for:
- Yamasa Digi World 4D -> Yamasa Digi World 4 (remove D)
- Shinkon Gattai Gondannar!! -> Shinkon Gattai Godannar!! (remove n)
- Correctly name all versions of Taiko no Tatsujin - Doka! to Oomori Nanadaime, Taiko no Tatsujin - Atsumare! Matsuri da!! Yondaime and Taiko no Tatsujin - Taiko Drum Master

- Add FMV fixes for the Yamasa Digi World 4 games.
- Add upscaling and deinterlacing fixes for the Taiko no Tatsujin and Taiko Drum Master games.

### Rationale behind Changes
- Add missing games, correct and unify naming of existing games.
- The two Yamasa Digi World 4 games need the SoftwareRendererFMVHack. Without it they display an epilepsy inducing flickering and random garbage pixels during the intro video.
- The Taiko no tatsujin games and Taiko Drum Master show vertical black lines when upscaling without the alignSprite fix and the notes are very blurry when deinterlacing is set to auto. This is fixed with forcing Bob (Bottom Field First) deinterlacing.

Taiko no tatsujin without alignSprite and with auto deinterlace
![gs_20220609211606_Taiko no Tatsujin - Atsumare_ Matsuri da__ Yondaime_SLPS-20383](https://user-images.githubusercontent.com/2962364/172936882-1072fbd7-2692-4804-bb88-52b61748ffa4.jpg)

Taiko no tatsujin with alignSprite enabled and deinterlace: 4
![gs_20220609212835_Taiko no Tatsujin - Atsumare_ Matsuri da__ Yondaime_SLPS-20383](https://user-images.githubusercontent.com/2962364/172937016-8ade1da8-921a-4f58-b402-498c9ac78f30.jpg)

refractionpcsx2 commented:
Just to check, those games you changed the deinterlace mode on don't bounce right?

noigeaR commented:
> Just to check, those games you changed the deinterlace mode on don't bounce right?
>
> Looks like you need to fix your formatting. Check for accidental spaces at the end or using tabs instead of spaces, etc.

The games don't bounce. They look stable in motion with bob bff and don't blur the notes like auto does.

I've re-checked and I have 2 spaces per indent for every entry I've added, no tabs and no trailing spaces.
I'm not sure what else I could check. Complete noob here, sorry. I might need some hand-holding to fix this.

Incidentally, I actually found 5 trailing spaces in GameIndex.yaml, but they already are in master and far away from any entries I've made.
They're at the end of comments for SLPM-65209, SLUS-20758 and SLUS-20759.

refractionpcsx2 commented on bin/resources/GameIndex.yaml:
```diff
@@ -32284,13 +32297,23 @@ SLPS-20250:
+  name: "Yamasa Digi World 4"
   region: "NTSC-J"
+  gameFixes:
+    - SoftwareRendererFMVHack  # Fix flickering and bad textures in FMV.
```
Double space before #

refractionpcsx2 commented on bin/resources/GameIndex.yaml:
```diff
@@ -32284,13 +32297,23 @@ SLPS-20250:
+  name: "Yamasa Digi World 4DX"
+  region: "NTSC-J"
+  gameFixes:
+    - SoftwareRendererFMVHack  # Fix flickering and bad textures in FMV.
```
Double space before #

refractionpcsx2 commented:
if it's not that, I'm not sure what it's bitching about :) I guess I'll have to get out the fine toothed comb

noigeaR commented:
Looks like I need another coffee...
I've corrected the 2 double spaces in my entries and also removed the 5 trailing spaces in the other entries while I was at it.

refractionpcsx2 commented:
Thanks, all looks good now
